### PR TITLE
Mark previous VCL files as "cold" when re-loading

### LINKED ISSUE #14: {{ range .Backends -}} doesn't de-register on pod delete
I've got a VCL running health checks:

```
{{ range .Backends }}
backend be-{{ .Name }} {
  .host = "{{ .Host }}";
  .port = "{{ .Port }}";
  .first_byte_timeout = 600s;
  .probe = {
      .url = "/health_check.php";
      .timeout = 2s;
      .interval = 5s;
      .window = 10;
      .threshold = 5;
  }
}
{{- end }}
```

Now even with 1 backend in the service to discover, we're getting multiple health checks (to the point it's overloading our server). (this is a single pod's logs here). I'm assuming it's been registered more than once in the range.Backends, but I could be wrong.

```
127.0.0.1 -  29/Jan/2020:23:40:46 +0000 "GET /health_check.php" 200
10.0.1.179:80 10.0.1.139 - - [29/Jan/2020:23:40:43 +0000] "GET /health_check.php HTTP/1.1" 200 517 "-" "-"
127.0.0.1 -  29/Jan/2020:23:40:46 +0000 "GET /health_check.php" 200
10.0.1.179:80 10.0.1.139 - - [29/Jan/2020:23:40:43 +0000] "GET /health_check.php HTTP/1.1" 200 517 "-" "-"
127.0.0.1 -  29/Jan/2020:23:40:46 +0000 "GET /health_check.php" 200
10.0.1.179:80 10.0.1.139 - - [29/Jan/2020:23:40:43 +0000] "GET /health_check.php HTTP/1.1" 200 517 "-" "-"
127.0.0.1 -  29/Jan/2020:23:40:46 +0000 "GET /health_check.php" 200
10.0.1.179:80 10.0.1.139 - - [29/Jan/2020:23:40:43 +0000] "GET /health_check.php HTTP/1.1" 200 517 "-" "-"
127.0.0.1 -  29/Jan/2020:23:40:46 +0000 "GET /health_check.php" 200
10.0.1.179:80 10.0.1.139 - - [29/Jan/2020:23:40:43 +0000] "GET /health_check.php HTTP/1.1" 200 517 "-" "-"
127.0.0.1 -  29/Jan/2020:23:40:46 +0000 "GET /health_check.php" 200
127.0.0.1 -  29/Jan/2020:23:40:46 +0000 "GET /health_check.php" 200
10.0.1.179:80 10.0.1.139 - - [29/Jan/2020:23:40:44 +0000] "GET /health_check.php HTTP/1.1" 200 517 "-" "-"
10.0.1.179:80 10.0.1.139 - - [29/Jan/2020:23:40:44 +0000] "GET /health_check.php HTTP/1.1" 200 517 "-" "-"
127.0.0.1 -  29/Jan/2020:23:40:46 +0000 "GET /health_check.php" 200
10.0.1.179:80 10.0.1.139 - - [29/Jan/2020:23:40:44 +0000] "GET /health_check.php HTTP/1.1" 200 517 "-" "-"
127.0.0.1 -  29/Jan/2020:23:40:46 +0000 "GET /health_check.php" 200
10.0.1.179:80 10.0.1.139 - - [29/Jan/2020:23:40:44 +0000] "GET /health_check.php HTTP/1.1" 200 517 "-" "-"
127.0.0.1 -  29/Jan/2020:23:40:46 +0000 "GET /health_check.php" 200
10.0.1.179:80 10.0.1.139 - - [29/Jan/2020:23:40:44 +0000] "GET /health_check.php HTTP/1.1" 200 517 "-" "-"
127.0.0.1 -  29/Jan/2020:23:40:46 +0000 "GET /health_check.php" 200
127.0.0.1 -  29/Jan/2020:23:40:46 +0000 "GET /health_check.php" 200
10.0.1.179:80 10.0.1.139 - - [29/Jan/2020:23:40:44 +0000] "GET /health_check.php HTTP/1.1" 200 517 "-" "-"
10.0.1.179:80 10.0.1.139 - - [29/Jan/2020:23:40:44 +0000] "GET /health_check.php HTTP/1.1" 200 517 "-" "-"
127.0.0.1 -  29/Jan/2020:23:40:46 +0000 "GET /health_check.php" 200
127.0.0.1 -  29/Jan/2020:23:40:46 +0000 "GET /health_check.php" 200
10.0.1.179:80 10.0.1.139 - - [29/Jan/2020:23:40:44 +0000] "GET /health_check.php HTTP/1.1" 200 517 "-" "-"
10.0.1.179:80 10.0.1.139 - - [29/Jan/2020:23:40:44 +0000] "GET /health_check.php HTTP/1.1" 200 517 "-" "-"
127.0.0.1 -  29/Jan/2020:23:40:46 +0000 "GET /health_check.php" 200
10.0.1.179:80 10.0.1.139 - - [29/Jan/2020:23:40:44 +0000] "GET /health_check.php HTTP/1.1" 200 517 "-" "-"
```

### PR DESCRIPTION
**Pull request by martin-helmich**
Fixes #14

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,7 +11,7 @@ require (
 	github.com/gregjones/httpcache v0.0.0-20180305231024-9cad4c3443a7 // indirect
 	github.com/imdario/mergo v0.3.6 // indirect
 	github.com/json-iterator/go v1.1.5 // indirect
-	github.com/martin-helmich/go-varnish-client v0.1.0
+	github.com/martin-helmich/go-varnish-client v0.1.1-0.20200302071804-9c9194146dfe
```
Comment (martin-helmich) [inline go.mod]: Reminder to self: This is a direct checkout of the dependent PR martin-helmich/go-varnish-client#5. Merge that one and update the dependence at this place again, before merging this one.

**Comment by martin-helmich** [comment]
Ugh, I was stupid. The much easier fix would have been to just create the new VCL files as `auto` instead of `warm`. This way, the `vcl_cooldown` variable will become effective and they will cool down automatically. 🤦‍♂

**Comment by martin-helmich** [comment]
@Hermsi1337 Mind giving this another look to see if we can get this merged? Don't let yourself be distracted by the failing Travis job, that was just me stupidly pushing into the wrong branch 🙈

**Comment by martin-helmich** [comment]
@Hermsi1337 Here's some more context for you ;)

**Comment by Hermsi1337** [comment]
Thats quite enough context for me 🥇

**Pull request merged**